parsec/core/fs/workspacefs.py

```diff
@@ -0,0 +1,138 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
+
+import math
+import inspect
+from uuid import UUID
+
+from parsec.core.types import FsPath
+from parsec.core.fs.local_folder_fs import FSManifestLocalMiss, FSMultiManifestLocalMiss
+from parsec.core.fs.local_file_fs import FSBlocksLocalMiss
+
+
+class WorkspaceFS:
+    def __init__(
+        self,
+        workpace_name,
+        device,
+        local_storage,
+        backend_cmds,
+        event_bus,
+        _local_folder_fs,
+        _local_file_fs,
+        _remote_loader,
+        _syncer,
+    ):
+        self.workpace_name = workpace_name
+        self.device = device
+        self.local_storage = local_storage
+        self.backend_cmds = backend_cmds
+        self.event_bus = event_bus
+        self._local_folder_fs = _local_folder_fs
+        self._local_file_fs = _local_file_fs
+        self._remote_loader = _remote_loader
+        self._syncer = _syncer
+
+    def _cook_path(self, relative_path=""):
+        return FsPath(f"/{self.workpace_name}/{relative_path}")
```

Comment:
We'll have to tackle this change of meaning for root at some point and get rid of the cooking